### ISSUE TITLE
Update the `mamba` integration

### ISSUE THREAD

**Issue by rlouf** [outlines-dev/outlines#808]
Similarly to #772 and #782 use logits processors. Also write the doc and see if we can add an example with a state space model.
